Improve table of contents

sarakonradi opened the pull request:
This PR updates the documentation's table of contents, making the list more skimmable to the user.

More specifically, it: 
- shortens long headings (max. 3 words)
- uses active verbs (e.g. "Install NEST")
- makes consistent use of sentence case for titles (e.g. "Model directory")
- provides a more logical order (e.g. "Getting help" and "Troubleshooting" at the bottom)
- makes subheadings more inviting (e.g. "Interested in contributing?")

terhorstd commented:
This could also be considered for #1445

clinssen commented:
In `doc/contents.rst`, `ref_material/nest3_vs_3` should read `ref_material/nest2_vs_3`

sarakonradi commented:
Thanks for noticing that @clinssen, just corrected it

heplesser commented on doc/contents.rst:
```diff
@@ -12,15 +12,15 @@ Table of Contents
-   ref_material/nest2_vs_3
    ref_material/pynest_apis
+   getting_help
+   troubleshooting
```
Maybe "troubleshooting" before "getting_help"?

sarakonradi replied:
Good idea! Done.

heplesser commented on doc/models/iaf.rst:
```diff
@@ -1,5 +1,5 @@
-Integrate-and-Fire (IF) neuron models
-==================================================
+Integrate-and-fire (IF) neuron models
```
IF -> IaF

sarakonradi replied:
This should be addressed in #1374

heplesser replied:
Hm, #1374 seems to remove `iaf.rst` .... We just need to keep an eye on it so once we have merged all the PRs, we have the correct end result.

heplesser commented on doc/ref_material/nest2_vs_3.rst:
```diff
@@ -1,7 +1,7 @@
-NEST 2.X vs. NEST 3.0 conversion reference guide
-==================================================
+NEST2-to-NEST3 transition guide
```
Should we really have the hyphens in there?

sarakonradi replied:
How about ``NEST 3.0 transition guide``? I suggest keeping the top level headings short, so that they fit on one line and the table of contents is skimmable for the user.

heplesser replied:
Good idea!

sarakonradi replied:
Done.

sarakonradi commented:
@heplesser Perhaps this PR should be marked as Blocked until #1475 is merged, since that affects the table of contents.

heplesser commented:
> @heplesser Perhaps this PR should be marked as Blocked until #1475 is merged, since that affects the table of contents.

@sarakonradi I would try to keep blocks to the absolute minimum, otherwise we will end in deadlock. Whichever one is merged second of these two will have to make necessary updates to ensure a consistent state in the end.

sarakonradi commented:
> @sarakonradi I would try to keep blocks to the absolute minimum, otherwise we will end in deadlock. Whichever one is merged second of these two will have to make necessary updates to ensure a consistent state in the end.

That is a fair point. Sure, we can make all updates after the first one is merged.

sarakonradi commented:
#1475 changed the conversion guide titles, and #1439 created two new documentation workflow pages.

I have now updated the table of contents and:
- Fixed the merge conflicts in ``doc/contents.rst``, ``doc/guides/nest2_to_nest3/nest2_to_nest3_detailed_transition_guide.rst`` and ``doc/index.rst``
- Included the recently created "Documentation workflows" in the navigation
- Renamed the overview guide to "NEST 3.0: What's new?", as discussed with @heplesser 

@clinssen @heplesser Could you please check if I have addressed everything?

terhorstd commented:
Maybe have a look if the word "NEST" is actually necessary in the titles.

clinssen commented on doc/troubleshooting.rst:
```diff
@@ -209,7 +206,7 @@ If your Python version is correct and you still have the same error, then try on
-----------------------------------------------
+--------------------------------------
 
 We may have answered your question on GitHub or in our Mailing List!
```
Suggest to make these hyperlinks.

sarakonradi replied:
That's a good suggestion! I updated the hyperlinks.

heplesser commented on doc/community.rst:
```diff
@@ -3,7 +3,7 @@
 NEST Community
 ===============
```
Is the underlining one character too long?

sarakonradi replied:
Yes, fixed it.

heplesser commented on doc/community.rst:
```diff
@@ -3,7 +3,7 @@
 
-Mailing List
+Mailing list
 -------------
```
Is the underlining one char too long?

sarakonradi replied:
Fixed it.

heplesser commented:
> Maybe have a look if the word "NEST" is actually necessary in the titles.

@terhorstd In the locations changed I did not notice any overuse of NEST in the titles. I don't think they hurt.

sarakonradi commented:
I now added a sentence about the documentation workflow to the main ``index.rst`` as well as ``contribute/index.rst``.

sarakonradi commented:
> @terhorstd In the locations changed I did not notice any overuse of NEST in the titles. I don't think they hurt.

@heplesser I think @terhorstd is referring to titles such as "Download NEST" and "Install NEST". Since these instructions are within the NEST simulator documentation, "NEST" is already implied to the user. I have attached a preview of what the navigation would look like without the extra "NEST" words. How do you find it?

![toctree_without_nest](https://user-images.githubusercontent.com/58212853/78763945-2285e400-7986-11ea-96d1-ce8d52a8e18c.png)
